### PR TITLE
Project intellij settings

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Add IntelliJ project settings to help setting up the IDE.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: IDE config

- [X] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by SchoolGuy** [comment]
@cbosdo Could you maybe add the non-Java settings as well. Or do we do this after we merged your Python refactoring? :)

**Comment by cbosdo** [comment]
> @cbosdo Could you maybe add the non-Java settings as well. Or do we do this after we merged your Python refactoring? :)

I don't have the non-Java settings since I'm using IntelliJ only for Java, but feel free to add them too

**Comment by SchoolGuy** [comment]
@cbosdo Will do. However it will take a while.

File: java/code/code.iml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/jackson-databind/jars/jackson-databind-2.10.5.1.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/salt-netapi-client/jars/salt-netapi-client-0.19.0.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/snakeyaml/jars/snakeyaml-1.28.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/pgjdbc-ng/jars/pgjdbc-ng-0.8.7.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/spy/jars/spy-0.8.7.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/client/jars/client-1.0.0~beta.2.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/common/jars/common-1.0.0~beta.2.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-c3p0/jars/hibernate-c3p0-5.3.25.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-core/jars/hibernate-core-5.3.25.jar!/" />
+          <root url="jar://$USER_HOME$/.ivy2/cache/suse/hibernate-ehcache/jars/hibernate-ehcache-5.3.25.jar!/" />
```
Comment (renner) [inline java/code/code.iml:149]: About this whole section I am not so sure: what if the versions are changing or differ locally, wouldn't this become a constant source of possible merge conflicts or at least unnecessary changes in patches? I guess that the ivy configuration file should probably be the only place where we define those version numbers checked into git.

Comment (cbosdo) [reply java/code/code.iml:149]: that is right... The issue is that some parts of that file are still needed to define the source tree (called `module` in IntelliJ's langage). I'll need to looks if something can be done for that

**Comment by SchoolGuy** [comment]
@nodeg Could you maybe add the configuration of the Testsuite. I think you showed me the testsuite in RubyMine and thus you could add the plugin to IntelliJ and then define the testsuite module as needed.

**Comment by nodeg** [comment]
I can remember that I showed you the workflow in RubyMine but I cannot remember that for IntelliJ.
